flax/nnx/bridge/module.py

```diff
@@ -123,7 +123,7 @@ def _module_meta_call(cls: type[M], *args, **kwargs) -> M:
   parent = None
   module: M
 
-  if parent_ctx is not None and parent_ctx.in_compact:
+  if parent_ctx is not None:
```

Comment:
Not sure we should go into this path when we're not inside `compact`. Maybe we could do the check bellow outside?

Reply:
Done.